lib/matplotlib/pyplot.py

```diff
@@ -1885,53 +1884,44 @@ def colormaps():
 
 def _setup_pyplot_info_docstrings():
     """
-    Generates the plotting and docstring.
+    Generates the plotting docstring.
 
     These must be done after the entire module is imported, so it is
     called from the end of this module, which is generated by
     boilerplate.py.
     """
-    # Generate the plotting docstring
-    import re
-
-    def pad(s, l):
-        """Pad string *s* to length *l*."""
-        if l < len(s):
-            return s[:l]
-        return s + ' ' * (l - len(s))
-
     commands = get_plot_commands()
 
     first_sentence = re.compile(r"(?:\s*).+?\.(?:\s+|$)", flags=re.DOTALL)
 
     # Collect the first sentence of the docstring for all of the
     # plotting commands.
     rows = []
-    max_name = 0
-    max_summary = 0
+    max_name = len("Function")
+    max_summary = len("Description")
     for name in commands:
         doc = globals()[name].__doc__
         summary = ''
         if doc is not None:
             match = first_sentence.match(doc)
             if match is not None:
-                summary = match.group(0).strip().replace('\n', ' ')
+                summary = inspect.cleandoc(match.group(0)).replace('\n', ' ')
         name = '`%s`' % name
         rows.append([name, summary])
         max_name = max(max_name, len(name))
         max_summary = max(max_summary, len(summary))
 
-    lines = []
     sep = '=' * max_name + ' ' + '=' * max_summary
-    lines.append(sep)
-    lines.append(' '.join([pad("Function", max_name),
-                           pad("Description", max_summary)]))
-    lines.append(sep)
-    for name, summary in rows:
-        lines.append(' '.join([pad(name, max_name),
-                               pad(summary, max_summary)]))
-    lines.append(sep)
-
+    lines = [
+        sep,
```

Comment:
A bit niggling, but I like readable names. :smile:. Can we make it `separator` or even `table_separator`?